version: "3"
x-podman:
  in-pod: false
networks:
  main: 
    external: false
services:
  gitea:
    container_name: gitea
    image: "{{ gitea_image }}:{{ gitea_image_version }}"
    user: "{{ gitea_user_uid }}"
    userns_mode: keep-id
    restart: always
    depends_on:
      gitea-db:
        condition: service_healthy
    networks:
      - main
    environment:
      - GITEA__database__DB_TYPE=postgres
      - GITEA__database__HOST=db:5432
      - GITEA__database__NAME=gitea
      - GITEA__database__USER={{ gitea_user }}
      - GITEA__database__PASSWD=gitea
    volumes:
      - ./data:/var/lib/gitea:Z
      - ./config:/etc/gitea:Z
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    ports:
      - "{{ gitea_web_port }}:3000"
      - "{{ gitea_ssh_port }}:2222"
    healthcheck:
      test: 
        - "CMD"
        - "curl"
        - "-f"
        - "http://127.0.0.1:80"
      interval: 30s
      timeout: 10s
      retries: 5
  gitea-db:
    container_name: gitea-db
    image: docker.io/postgres:16.3@sha256:d0f363f8366fbc3f52d172c6e76bc27151c3d643b870e1062b4e8bfe65baf609
    user: "{{ gitea_user_uid }}"
    userns_mode: keep-id
    restart: always
    networks:
      - main
    environment:
      - POSTGRES_USER={{ gitea_user }}
      - POSTGRES_PASSWORD=gitea
      - POSTGRES_DB=gitea
    volumes:
      - ./postgres:/var/lib/postgresql/data:Z
    healthcheck:
      test: 
        - CMD-SHELL
        - pg_isready
      interval: 10s
      timeout: 5s
      retries: 5
